src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -762,6 +764,7 @@ private async Task TryNotifyClientErrorAsync(IClientProxy client, string error,
 
                     // Retrieve the circuit handlers at this point.
                     _circuitHandlers = [.. _scope.ServiceProvider.GetServices<CircuitHandler>().OrderBy(h => h.Order)];
+                    _dispatchInboundActivity = BuildInboundActivityDispatcher(_circuitHandlers, Circuit);
```

Comment:
This is pretty odd - it's hard to see why we have to do this here in particular. Is it just a matter of lazy initialization? If so would it be clearer still if the field was a `Lazy<T>` that was always initialized during the constructor, and then we wouldn't need to initialize it explicitly in other places.

Reply:
I agree it's a bit confusing. I believe the reason we collect the circuit handlers here is that calling `_scope.GetServices<CircuitHandler>()` may construct new instances of services registered as circuit handlers, and those services may access persistent component state in their constructors. So we need to be sure that persistent component state is available by the time these services get constructed. We might be able to use a `Lazy<T>`, but my concern is that making circuit handlers lazily and implicitly initialized might cause us to accidentally change when circuit handlers are constructed without us knowing it.

Reply:
Yeah, this code is correct. The Blazor Server version also does the same thing.